tests/unit/transformers/test_base.py

```diff
@@ -28,16 +455,18 @@ def test__set_columns_data_preserves_order_series(self):
         columns = ['c']
```

Comment:
I think that the input dataframe should not contain the output column, since in the real usage when this method is called it will always be creating new columns. The same applies to the other variations of this test.